app/main.py

```diff
@@ -1,6 +1,21 @@
-from typing import Callable
+from typing import Callable, Any
+import functools
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    result_dict = {}
+
+    @functools.wraps(func)
+    def inner(*args: Any) -> Any:
```

Comment:
what if func takes key arguments?
don't forget about kwargs

Reply:
Also, What if two different func takes the same arguments?

Reply:
About second comment: give a hint please)) Where to look for the solution?  All tests are passed with in previous commit. Any other tries lead to failed tests.